lib/ddtrace/patcher.rb

```diff
@@ -21,21 +21,24 @@ def without_warnings
         end
       end
 
-      def do_once(key = nil)
+      def do_once(key = nil, options = {})
         # If already done, don't do again
         @done_once ||= {}
```

Comment:
You can remove line 34 if this uses the Hash initializer that takes a block:

```ruby
@done_once ||= Hash.new { |h, k| h[k] = {} }
```

I also wonder if a `Set` might be a better data structure for the values of the `@done_once` hash...

Reply:
This is a good suggestion, I will try this out.